protocol/streaming/full_node_streaming_manager.go

```diff
@@ -424,9 +523,47 @@ func (sm *FullNodeStreamingManagerImpl) SendTakerOrderStatus(
 	)
 }
 
-// AddUpdatesToCache adds a series of updates to the full node streaming cache.
+// SendSubaccountUpdates groups subaccount updates by their subaccount ids and
+// sends messages to the subscribers.
+func (sm *FullNodeStreamingManagerImpl) SendSubaccountUpdates(
+	subaccountUpdates []satypes.StreamSubaccountUpdate,
+	blockHeight uint32,
+	execMode sdk.ExecMode,
+) {
+	defer metrics.ModuleMeasureSince(
+		metrics.FullNodeGrpc,
+		metrics.GrpcSendSubaccountUpdatesLatency,
+		time.Now(),
+	)
+
+	// Group subaccount updates by subaccount id.
+	streamUpdates := make([]clobtypes.StreamUpdate, 0)
+	subaccountIds := make([]*satypes.SubaccountId, 0)
+	for _, subaccountUpdate := range subaccountUpdates {
```

Comment:
We could eliminate the grouping logic by making it take in one subaccount update, same as above comment

Reply:
StreamUpdate is 1:1 with SubaccountUpdate so we will still need to send multiple StreamUpdates. But yes, if we implement above (added a todo for now), we will only have 1 SubaccountUpdate per subaccountid.